packages/zosfiles/src/cli/rename/dsm/Dsm.definition.ts

```diff
@@ -0,0 +1,55 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { ICommandDefinition } from "@zowe/imperative";
+
+import i18nTypings from "../../-strings-/en";
+
+// Does not use the import in anticipation of some internationalization work to be done later.
+const strings = (require("../../-strings-/en").default as typeof i18nTypings);
+const dsmStrings = strings.RENAME.ACTIONS.DATA_SET_MEMBER;
+
+export const DsmDefinition: ICommandDefinition = {
+    name: "data-set-member",
+    aliases: ["dsm"],
+    description: dsmStrings.DESCRIPTION,
+    type: "command",
+    handler: __dirname + "/dsm.handler",
+    profile: {
+        optional: ["zosmf"]
+    },
+    positionals: [
+        {
+            name: "dataSetName",
+            type: "string",
+            description: dsmStrings.POSITIONALS.DSNAME,
+            required: true
+        },
+        {
+            name: "beforeMemberName",
+            type: "string",
+            description: dsmStrings.POSITIONALS.BEFOREMEMBERNAME,
+            required: true
+        },
```

Comment:
I think there are examples of doing both `dataset(member)` and `dataset member` or having member as another option... so consistency is not going to help us... but for what it's worth, if you choose to accept `dataset(member)` you'll have to deconstruct it for the API (our rename class and the z/OSMF endpoint). Looks like z/OSMF handles it as [separate entries in the URI and payload](https://github.com/zowe/zowe-cli/blob/10b34e5c73620e642cf989025d2a97d3d9e2439f/packages/zosfiles/src/api/methods/rename/Rename.ts#L83). Plus, the way it is now, you can at least validate via imperative definition that all three are specified.

Reply:
Good point about consistency not helping here.
Maybe we can have an issue to address consistency in the `@next` version.

In the meantime, I don't think it's a problem to have 3 positional args. And it might make sense to have 3 to avoid adding more possible failure points (parsing ds name ourselves)